### PR TITLE
Dev/release 1.1.20

### PR DESCRIPTION
**Pull request by Ishika1601-boxpay**
it includes
1. Shopper enabled flag
2. Otp Configurable
3. Overall sdk bugs 
4. DCC added
5. Pan and Dob Added
6. Added device details and clarity extra tag
7. added the timer in which the session will expire automatically instead of any api call
8. EMI flow
9. Swipe to pay button
10. Success screen visibility on conditions
11. Recurring payment (Subscription)
12. Upi Stand Alone

### CODE REVIEW

File: Sdk/src/main/java/com/boxpay/checkout/sdk/AddCardBottomSheet.kt [hunk 1]
```diff
@@ -727,6 +865,121 @@ internal class AddCardBottomSheet : BottomSheetDialogFragment() {
...
+                    decoderFactory { result, options, _ -> SvgDecoder(result.source, options) }
+                    scale(Scale.FIT) // Ensures the image fits within the ImageView
+                }
+                isDCCFetched = true
+            }
+
+        }else{
+            binding.tvInfoDcc.text = ""
+            isDCCFetched = false
+            isDCCEnabled = true
```
Comment (ankushboxpay) [inline Sdk/src/main/java/com/boxpay/checkout/sdk/AddCardBottomSheet.kt:978]: h

File: Sdk/src/main/java/com/boxpay/checkout/sdk/DeliveryAddressBottomSheet.kt [hunk 2]
```diff
@@ -286,25 +250,28 @@ class DeliveryAddressBottomSheet : BottomSheetDialogFragment() {
         binding.spinnerDialCodes.setOnItemClickListener { parent, view, position, id ->
             val selectedDialCode = parent.getItemAtPosition(position).toString()
             // Display or use the selected item
```
Comment (ankushboxpay) [inline Sdk/src/main/java/com/boxpay/checkout/sdk/DeliveryAddressBottomSheet.kt:252]: a

File: Sdk/src/main/java/com/boxpay/checkout/sdk/DeliveryAddressBottomSheet.kt [hunk 3]
```diff
@@ -286,25 +250,28 @@ class DeliveryAddressBottomSheet : BottomSheetDialogFragment() {
...
-            }
-            if (isMobileNumberValid()) {
-                enableProceedButton()
+            if (!selectedDialCode.contains("no",true)) {
+                countryCodePhoneNum = selectedDialCode
+                indexCountryCodePhone = selectedDialCode
+                phoneCodeSelected = true
+                countrySelectedFromDropDown =
+                    setCountryNameUsingPhoneCode(countryCodeJson, countryCodePhoneNum)
+                binding.countryEditText.setText(countrySelectedFromDropDown)
```
Comment (ankushboxpay) [inline Sdk/src/main/java/com/boxpay/checkout/sdk/DeliveryAddressBottomSheet.kt:259]: k

File: Sdk/src/main/java/com/boxpay/checkout/sdk/AddCardBottomSheet.kt [hunk 4]
```diff
@@ -374,13 +496,31 @@ internal class AddCardBottomSheet : BottomSheetDialogFragment() {
...
-                                requireContext(),
-                                text.substring(0, 9)
+                                requireContext(), text.substring(0, 9), text
                             )
-                        else {
+                            isCardNumberValid = true
+
+                        } else {
                             binding.ll1InvalidCardNumber.visibility = View.INVISIBLE
+                            isCardNumberValid = false
```
Comment (ankushboxpay) [inline Sdk/src/main/java/com/boxpay/checkout/sdk/AddCardBottomSheet.kt:507]: d

File: app/src/main/java/com/boxpay/checkout/demoapp/Check.kt [hunk 5]
```diff
@@ -98,161 +98,165 @@ class Check : AppCompatActivity() {
 
     private fun makePaymentRequest(context: Context) {
         val queue = Volley.newRequestQueue(context)
-        val url = "https://test-apis.boxpay.tech/v0/merchants/lGfqzNSKKA/sessions"
```
Comment (ankushboxpay) [inline app/src/main/java/com/boxpay/checkout/demoapp/Check.kt:101]: d

File: Sdk/src/main/java/com/boxpay/checkout/sdk/AddCardBottomSheet.kt [hunk 6]
```diff
@@ -1065,6 +1354,17 @@ internal class AddCardBottomSheet : BottomSheetDialogFragment() {
                 put("lastName", sharedPreferences.getString("lastName", null))
                 put("phoneNumber", sharedPreferences.getString("phoneNumber", null))
                 put("uniqueReference", sharedPreferences.getString("uniqueReference", null))
+                if (sharedPreferences.getString("dateOfBirthChosen", null) != null){
+                    put("dateOfBirth", sharedPreferences.getString("dateOfBirthChosen", null))
```
Comment (ankushboxpay) [inline Sdk/src/main/java/com/boxpay/checkout/sdk/AddCardBottomSheet.kt:1358]: f

File: Sdk/src/main/java/com/boxpay/checkout/sdk/MainBottomSheet.kt [hunk 7]
```diff
@@ -711,6 +779,12 @@ internal class MainBottomSheet : BottomSheetDialogFragment(), UpdateMainBottomSh
...
                     val errorResponse = String(error.networkResponse.data)
                     val errorMessage = extractMessageFromErrorResponse(errorResponse)
+                    callUIAnalytics(
+                        requireActivity(),
+                        "ERROR_GETTING_UPI_URL $errorMessage",
+                        "",
+                        "UPI"
+                    )
 
                     if (errorMessage?.contains("expired", true) == true) {
```
Comment (ankushboxpay) [inline Sdk/src/main/java/com/boxpay/checkout/sdk/MainBottomSheet.kt:789]: h

**Review by ankushboxpay — DISMISSED** [review]
Reviewed Successfully

**Review by aksain-boxpay — APPROVED** [review]

**Pull request merged**